### ISSUE TITLE
APSIMX vs 

### ISSUE THREAD

**Issue by hut104** [APSIMInitiative/ApsimX#4690]

**Comment by sarahcleary** [comment]
This question was discussed in the RP meeting.  ACTION:  @APSIMInitiative to seek feedback on implications on renaming (1) the repository from APSIMX to APSIMNG and the file extension to .apsimng.  @hol430; @hol353 - others - any ideas?

**Comment by peter-devoil** [comment]
[They say](https://help.github.com/en/github/administering-a-repository/renaming-a-repository) it is easy to rename a repo, and indicate that old scripts that use the old url will be redirected to the new one.

**Comment by sarahcleary** [comment]
@peter-devoil; @HamishBrownPFR - are you happy with proceeding with renaming the repository?  and subsequently, the file extension?

**Comment by sarahcleary** [comment]
@peter-devoil; @HamishBrownPFR - next step/s?

**Comment by peter-devoil** [comment]
Wait for a "quiet time", and ask @hol430 to implement it. I expect that most users will be unaffected, but whether the jenkins scripts will handle the redirects is unknown right now.

**Comment by hol430** [comment]
Yes, I can look into this at some point. I think renaming the repository shouldn't be a big deal, the github docs you linked earlier say that there will be a redirect in place. We'll still want to update the jenkins scripts and other web services of course.

Changing the file extension sounds like a great way to confuse people but should be easy enough to implement (my favourite sort of change).

**Comment by hol353** [comment]
I agree with @hol430 - changing the file extension will confuse users. How will they know whether a file is for APSIM 7.10 or APSIM Next Gen?

**Comment by hut104** [comment]
Perhaps having the current extension implying XML is similarly unhelpful.
Do we just be explicit and use *.apsimng or similar?

**Comment by hol353** [comment]
Yep that would work.

**Comment by peter-devoil** [comment]
What's wrong with .apsim?

**Comment by hut104** [comment]
As raised above, that extension is already used for classic.

**Comment by lie112** [comment]
What about allowing apsimx, apsimng to be loaded from the open simulation box for now. The user will be presented with an error if the file format is not suitable. From now on application saves and expects all simulations as apsimng. This will allow user transition. Maybe the previously opened list could contain apsimx files until an apsimng file is saved and then the apsimx is removed from previously opened list and deleted from HDD. This would be a similar approach to the rollout of json version of apsimx files. There wouldn't be any confusion with APSIM classic.

**Comment by hol353** [comment]
So if we're not changing the file extension then the last remaining job is to rename the repo. This will mean that everyone will have to update their git client software to point to the new URL. Anything that links to https://github.com/APSIMInitiative/ApsimX will no longer work. 

Are you sure you want to do this?

**Comment by peter-devoil** [comment]
If your software handles redirects, there shouldn't be a problem. That includes git, tortoisegit, wget etc

Test in a sandbox to be sure?

**Comment by hut104** [comment]
I didn't know we weren't changing the extension.

**Comment by sarahcleary** [comment]
Sorry, finalizing the notes from the last RP meeting.  Draft notes:  @APSIMInitiative/reference-panel  agreed that although renaming the extension from .apsimx to .apsimng would help with communication, it may cause some errors/issues which will require fixing. At this stage, agreed for the extension to stay as .apsimx and utilise the NG in ALL other communication. Agreed to look at renaming the Repository for communication purposes.

**Comment by sarahcleary** [comment]
@hol353; @peter-devoil - next step/s?

**Comment by peter-devoil** [comment]
Wait for a "quiet time"...
